src/dom/Draggable.js

```diff
@@ -112,9 +111,12 @@ export var Draggable = Evented.extend({
 		// Fired when a drag is about to start.
 		this.fire('down');
 
-		var first = e.touches ? e.touches[0] : e;
+		var first = e.touches ? e.touches[0] : e,
+		    sizedParent = DomUtil.getSizedParentNode(this._element);

```

Comment:
Sorry for taking so long to review this PR :)

Do we need to look for sized parent node *every* time `_onDown` or `_onMove` are fired?
Unless I'm mistaken, it's enough to do that once in `initialize` method

Reply:
Hi,

Thank you for your review!

Well, there is the edge case where container scale is changed dynamically…

A more common case is when the map container has an initially null size (therefore leading to usage of `invalidateSize` later on).
Since `Draggable` is initialized on the map initialization, it would not be able to compute the scaled size at that moment.

Would be happy to find a suitable way to cache the scale though indeed, but covering as much use cases as possible and providing a workaround for other cases…

Reply:
Maybe cache on `_onDown`, hoping that the scale would not change for the drag duration?

Reply:
>Well, there is the edge case where container scale is changed dynamically…

I don't think it's a problem - we will store only *reference* to that element and we'll get it's size on each `_onDown` and `_onMove`

Reply:
>A more common case is when the map container has an initially null size (therefore leading to usage of invalidateSize later on).

Huh, that may be a problem

>Since Draggable is initialized on the map initialization, it would not be able to compute the scaled size at that moment.

The real problem is that in case of null size map container it will skip it because of null size :/

Reply:
What if we replace `getSizedParentNode` with `getMapContainer` function (like [that](https://github.com/Leaflet/Leaflet/pull/5944/files#diff-b2be552947d40375eca631cc4a3cd4ddR44))?
This way we can store reference to map container during `initialize` method and read it's size on each `onDown/onMove`

Reply:
> we will store only _reference_ to that element and we'll get it's size on each `_onDown` and `_onMove`

True, looks like I already forgot how this PR works… :sweat_smile:

> What if we replace `getSizedParentNode` with `getMapContainer` function?

That is interesting indeed. The advantage is that the container detection is only done once, at instantiation. But now `Draggable` class becomes aware of map characteristics, whereas until now [it can be used independently from any map](https://github.com/Leaflet/Leaflet/pull/5997#discussion_r167515814). Same for `DomUtil`.

I would tend to keep `getSizedParentNode` (to keep `Draggable` agnostic) and to cache the found container in `_onDown`. That would avoid repeatedly looking for that container in `_onMove`. We can reasonably assume that if a drag is started, then there is a non null size parent container. If its scale changes dynamically, it will be accounted for in `getMousePosition` (but a further optimization could also cache the scale in `_onDown`, assuming it will not change during the drag, or user will restart their drag anyway due to the disruption in the map display when its scale changes).

IMHO I do not think looking for the container on every `_onDown` would be such a performance issue: it is just walking up the DOM parent hierarchy. And `_onDown` is triggered on user drag start, much less often than `_onMove`.

What do you think?

Reply:
> But now Draggable class becomes aware of map characteristics, whereas until now it can be used independently from any map.

You are right, let's keep `Draggable` agnostic!

So we have two options:
- cache sized container
- cache scale

My vote is for caching scale, but I don't really think there's significant performance difference between those approaches.
The important thing is that either way `_onMove` will be much faster :)

Reply:
Thanks for sharing that link, didn't know about that!

But in Appendix section it says:
> Reflow only has a cost if the document has changed and invalidated the style or layout

So I'm wondering if reflow has a cost in our case

Reply:
@ghybs I agree ;)